app/views/TaxYearOverview.scala.html

```diff
@@ -73,48 +73,55 @@
-                        } else {
-                            <a class="govuk-link"  id="paymentTypeLink-@{index}" href="@getChargeSummaryHref(documentDetailWithDueDate)">@messages(s"tax-year-overview.payments.${documentDetailWithDueDate.documentDetail.getChargeTypeKey(codingOutEnabled)}")</a>
+@h2(msg = messages("tax-year-overview.payments"), classes = "govuk-heading-l")
+@if(charges.isEmpty){
+    @p()(content = Html(messages("tax-year-overview.payments.no-payments")))
+} else {
+    <table class="govuk-table" id="payments-table">
+        <caption class="govuk-table__caption govuk-visually-hidden" style="margin-bottom: 0px;">@messages("tax-year-overview.payments")</caption>
+        @paymentsTableHead("payments-table-head")
+        <tbody>
+        @for((documentDetailWithDueDate, index) <- charges.sortBy(_.dueDate.getOrElse(throw MissingFieldException("Due Date"))).sortBy(_.documentDetail.isPayeSelfAssessment).zipWithIndex) {
+            <tr class="govuk-table__row">
+                <td class="govuk-table__cell">
+                    @if(documentDetailWithDueDate.isOverdue && !documentDetailWithDueDate.documentDetail.isPayeSelfAssessment && ((!documentDetailWithDueDate.isLatePaymentInterest && !documentDetailWithDueDate.documentDetail.isPaid) || (documentDetailWithDueDate.isLatePaymentInterest && !documentDetailWithDueDate.documentDetail.interestIsPaid))) {
+                        <div class="govuk-tag govuk-tag--red">@messages("tax-year-overview.payments.overdue")</div>
+                    }
+                <div>
+                    @if(documentDetailWithDueDate.isLatePaymentInterest) {
+                        <a class="govuk-link"  id="paymentTypeLink-@{index}" href="@getChargeSummaryHref(documentDetailWithDueDate, true)">@messages(s"tax-year-overview.payments.lpi.${documentDetailWithDueDate.documentDetail.getChargeTypeKey()}")</a>
+                    } else {
+                        <a class="govuk-link"  id="paymentTypeLink-@{index}" href="@getChargeSummaryHref(documentDetailWithDueDate)">@messages(s"tax-year-overview.payments.${documentDetailWithDueDate.documentDetail.getChargeTypeKey(codingOutEnabled)}")</a>
+                    }
+                </div>
+                    @if(documentDetailWithDueDate.dunningLock){<div class="form-hint govuk-body-s"> @messages("tax-year-overview.payments.paymentUnderReview") </div>}
+                    @if(documentDetailWithDueDate.documentDetail.hasLpiWithDunningBlock){<div class="form-hint govuk-body-s"> @messages("tax-year-overview.payments.paymentUnderReview") </div>}
+                </td>
+                @if(documentDetailWithDueDate.isLatePaymentInterest) {
+                    @documentDetailWithDueDate.documentDetail.latePaymentInterestAmount.map { latePaymentInterestAmount =>
+                        @documentDetailWithDueDate.dueDate.map { dueDate =>
```

Comment:
Sorry missed this one in first review. This dueDate needs to throw an exception as well.